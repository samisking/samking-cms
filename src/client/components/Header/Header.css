.wrapper {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 60px;
  z-index: 3;
}

.header {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100vh;
  padding: 20px 30px;
  transform: translate3d(-50%, -50%, 0) rotate(-90deg);
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.navigation {
  display: flex;
}

.navigationItem {
  & + & {
    margin-left: 1em;
  }
}

.navigationLink {
  opacity: 0.3;
}

.navigationLink_active {
  composes: navigationLink;
  text-decoration: underline;
}
