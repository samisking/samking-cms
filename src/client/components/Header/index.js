import React, { PropTypes } from 'react';
import Link from 'react-router/lib/Link';
import pureRender from '../PureRender';
import styles from './Header.css';

const NavigationLink = ({ url, title, pathname = '' }) => {
  const className = pathname.startsWith(url) ?
    styles.navigationLink_active :
    styles.navigationLink;

  return (
    <div className={styles.navigationItem}>
      <Link to={url} className={className}>{title}</Link>
    </div>
  );
};

NavigationLink.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  pathname: PropTypes.string
};

const Header = ({ navigationLinks, currentPathname }) =>
  <div className={styles.wrapper}>
    <div className={styles.header}>
      <NavigationLink
        key={'logout'}
        url={'/logout'}
        title={'Logout'}
      />

      <div className={styles.navigation}>
        {navigationLinks.map(link =>
          <NavigationLink
            key={link.url}
            pathname={currentPathname}
            {...link}
          />
        )}
      </div>
    </div>
  </div>;

Header.propTypes = {
  navigationLinks: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired
    })
  ).isRequired,
  currentPathname: PropTypes.string
};

export default pureRender()(Header);
