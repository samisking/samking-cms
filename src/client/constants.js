export const navigationLinks = [
  {
    title: 'Photos',
    url: '/photos'
  },
  {
    title: 'Design',
    url: '/design'
  }
];
