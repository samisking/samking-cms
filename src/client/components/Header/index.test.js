import React from 'react';
import renderer from 'react-test-renderer';
import { navigationLinks } from '../../constants';
import Header from '.';

const basicProps = {
  navigationLinks,
  currentPathname: '/photos'
};

describe('<Header />', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Header {...basicProps} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
