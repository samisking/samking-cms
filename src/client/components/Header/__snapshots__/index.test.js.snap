exports[`<Header /> renders correctly 1`] = `
<div
  className="wrapper">
  <div
    className="header">
    <a
      className="brand"
      onClick={[Function]}
      style={Object {}}>
      Sam King
    </a>
    <div
      className="navigation">
      <div
        className="navigationItem">
        <a
          className="navigationLink_active"
          onClick={[Function]}
          style={Object {}}>
          Photos
        </a>
      </div>
      <div
        className="navigationItem">
        <a
          className="navigationLink"
          onClick={[Function]}
          style={Object {}}>
          Design
        </a>
      </div>
    </div>
  </div>
</div>
`;
